### ISSUE TITLE
unicode decode error when viewing bundle that's running

### ISSUE THREAD

**Issue by percyliang** [codalab/codalab-worksheets#1387]
Result of the Python 3 migration.  To reproduce, just start a running bundle like `while true; do date; sleep 1; done` and select the bundle.  It generates messages like: ```Error on request by pliang(0x5ed2416a59e041aeab72c991ac12020e):

GET /bundles/0x425ee7c0f4af4071a04bba42f1baef49/contents/blob/stderr

Query params:
head='50'
tail='50'
truncation_text='\n... [truncated] ...\n\n'
_='1568171042550'

Form params:

JSON body:
<none>

Local variables:
fileobj=<_io.BufferedReader name=-1>
response_socket_id=967```

**Comment by epicfaace** [comment]
rest of the stack trace:

```
self=<codalab.lib.download_manager.DownloadManager object at 0x7f06db413cc0>

Traceback (most recent call last):
  File "/opt/codalab-worksheets/codalab/server/rest_server.py", line 147, in wrapper
    return callback(*args, **kwargs)
  File "/usr/local/bin/bottle.py", line 1732, in wrapper
    rv = callback(*a, **ka)
  File "/opt/codalab-worksheets/codalab/server/json_api_plugin.py", line 18, in wrapper
    result = callback(*args, **kwargs)
  File "/opt/codalab-worksheets/codalab/rest/bundles.py", line 570, in _fetch_bundle_contents_blob
    uuid, path, head_lines, tail_lines, max_line_length, truncation_text, gzipped_stream
  File "/opt/codalab-worksheets/codalab/lib/download_manager.py", line 21, in wrapper
    return f(*args, **kwargs)
  File "/opt/codalab-worksheets/codalab/lib/download_manager.py", line 209, in summarize_file
    string = self._get_read_response_string(response_socket_id)
  File "/opt/codalab-worksheets/codalab/lib/download_manager.py", line 291, in _get_read_response_string
    return fileobj.read().decode()
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
```

**Comment by epicfaace** [comment]
The issue is this line of code: https://github.com/codalab/codalab-worksheets/blob/5b9390b5ecdead2ef736c4e43dc94037f2aeb161/codalab/lib/download_manager.py#L291

(In python 2, we didn't decode the data).

From [this link](https://stackoverflow.com/questions/44659851/unicodedecodeerror-utf-8-codec-cant-decode-byte-0x8b-in-position-1-invalid/44660123), it looks like "0x8b" byte is indicative of a gzipped file. We probably shouldn't be decoding the file here. I must have assumed that `_get_read_response_string` returns a string, so it must have `.decode()` in it; that assumption was wrong.

**Comment by percyliang** [comment]
Fixed via #1389
